### ISSUE TITLE
[🐛BUG] Abnormal results for HGN

### ISSUE THREAD

**Issue by Li-Muyang** [RUCAIBox/RecBole#1130]
Hi all, I was running HGN (Hierarchical Gating Networks for Sequential Recommendation) but the experiment results seems abnormal, is my configuration violates the original settings somehow? 

`26 Jan 02:58    INFO  
General Hyper Parameters:
gpu_id = 0
use_gpu = True
seed = 2020
state = INFO
reproducibility = True
data_path = 
show_progress = False
save_dataset = False
save_dataloaders = False
benchmark_filename = None

Training Hyper Parameters:
checkpoint_dir = saved
epochs = 100
train_batch_size = 256
learner = adam
learning_rate = 0.0001
eval_step = 1
stopping_step = 10
clip_grad_norm = None
weight_decay = 0.0
loss_decimal_place = 4

Evaluation Hyper Parameters:
eval_args = {'split': {'LS': 'valid_and_test'}, 'order': 'TO', 'mode': 'pop100', 'group_by': 'user'}
metrics = ['Recall', 'MRR', 'NDCG', 'Hit', 'Precision']
topk = [10]
valid_metric = MRR@10
valid_metric_bigger = True
eval_batch_size = 512
metric_decimal_place = 4

Dataset Hyper Parameters:
field_separator = 	
seq_separator =  
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = rating
threshold = None
NEG_PREFIX = neg_
load_col = {'inter': ['user_id', 'item_id', 'rating', 'timestamp'], 'item': ['item_id', 'class', 'movie_title', 'release_year']}
unload_col = None
unused_col = None
additional_feat_suffix = None
rm_dup_inter = None
val_interval = None
filter_inter_by_user_or_item = True
user_inter_num_interval = [20,inf)
item_inter_num_interval = [15,inf)
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = True
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 200
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id

Other Hyper Parameters: 
neg_sampling = {'uniform': 1}
repeatable = True
embedding_size = 64
loss_type = BPR
pooling_type = average
reg_weight = [0.0, 0.0]
MODEL_TYPE = ModelType.SEQUENTIAL
MODEL_INPUT_TYPE = InputType.PAIRWISE
eval_type = EvaluatorType.RANKING
device = cuda
train_neg_sample_args = {'strategy': 'by', 'by': 1, 'distribution': 'uniform'}
eval_neg_sample_args = {'strategy': 'by', 'by': 100, 'distribution': 'popularity'}

26 Jan 02:58    INFO  ml-1m
The number of users: 6031
Average actions of users: 165.3127694859038
The number of items: 3134
Average actions of items: 318.172997127354
The number of inters: 996836
The sparsity of the dataset: 94.72605746717899%
Remain Fields: ['user_id', 'item_id', 'rating', 'timestamp', 'movie_title', 'release_year', 'class']
26 Jan 02:59    INFO  [Training]: train_batch_size = [256] negative sampling: [{'uniform': 1}]
26 Jan 02:59    INFO  [Evaluation]: eval_batch_size = [512] eval_args: [{'split': {'LS': 'valid_and_test'}, 'order': 'TO', 'mode': 'pop100', 'group_by': 'user'}]
26 Jan 02:59    INFO  HGN(
  (item_embedding): Embedding(3134, 64, padding_idx=0)
  (user_embedding): Embedding(6031, 64)
  (w1): Linear(in_features=64, out_features=64, bias=True)
  (w2): Linear(in_features=64, out_features=64, bias=True)
  (w3): Linear(in_features=64, out_features=1, bias=False)
  (w4): Linear(in_features=64, out_features=200, bias=False)
  (item_embedding_for_prediction): Embedding(3134, 64)
  (sigmoid): Sigmoid()
  (loss_fct): BPRLoss()
)
Trainable parameters: 808320
26 Jan 02:59    INFO  epoch 0 training [time: 24.54s, train loss: 437.8752]
26 Jan 02:59    INFO  epoch 0 evaluating [time: 5.76s, valid_score: 0.030400]
26 Jan 02:59    INFO  valid result: 
recall@10 : 0.1005    mrr@10 : 0.0304    ndcg@10 : 0.0464    hit@10 : 0.1005    precision@10 : 0.01    
26 Jan 02:59    INFO  Saving current best: saved/HGN-Jan-26-2022_02-59-10.pth
26 Jan 03:00    INFO  epoch 1 training [time: 21.90s, train loss: 15.5824]
26 Jan 03:00    INFO  epoch 1 evaluating [time: 5.76s, valid_score: 0.027300]
26 Jan 03:00    INFO  valid result: 
recall@10 : 0.097    mrr@10 : 0.0273    ndcg@10 : 0.0432    hit@10 : 0.097    precision@10 : 0.0097    
26 Jan 03:00    INFO  epoch 2 training [time: 21.79s, train loss: 1.8377]
26 Jan 03:00    INFO  epoch 2 evaluating [time: 5.82s, valid_score: 0.027900]
26 Jan 03:00    INFO  valid result: 
recall@10 : 0.0987    mrr@10 : 0.0279    ndcg@10 : 0.044    hit@10 : 0.0987    precision@10 : 0.0099    
26 Jan 03:00    INFO  epoch 3 training [time: 21.71s, train loss: 0.2770]
26 Jan 03:01    INFO  epoch 3 evaluating [time: 5.75s, valid_score: 0.027300]
26 Jan 03:01    INFO  valid result: 
recall@10 : 0.096    mrr@10 : 0.0273    ndcg@10 : 0.0429    hit@10 : 0.096    precision@10 : 0.0096    
26 Jan 03:01    INFO  epoch 4 training [time: 21.99s, train loss: 0.0473]
26 Jan 03:01    INFO  epoch 4 evaluating [time: 5.83s, valid_score: 0.027400]
26 Jan 03:01    INFO  valid result: 
recall@10 : 0.1013    mrr@10 : 0.0274    ndcg@10 : 0.0442    hit@10 : 0.1013    precision@10 : 0.0101    
26 Jan 03:01    INFO  epoch 5 training [time: 21.68s, train loss: 0.0090]
26 Jan 03:01    INFO  epoch 5 evaluating [time: 5.78s, valid_score: 0.027000]
26 Jan 03:01    INFO  valid result: 
recall@10 : 0.097    mrr@10 : 0.027    ndcg@10 : 0.0429    hit@10 : 0.097    precision@10 : 0.0097    
26 Jan 03:02    INFO  epoch 6 training [time: 21.95s, train loss: 0.0021]
26 Jan 03:02    INFO  epoch 6 evaluating [time: 5.77s, valid_score: 0.026800]
26 Jan 03:02    INFO  valid result: 
recall@10 : 0.101    mrr@10 : 0.0268    ndcg@10 : 0.0436    hit@10 : 0.101    precision@10 : 0.0101    
26 Jan 03:02    INFO  epoch 7 training [time: 21.78s, train loss: 0.0006]
26 Jan 03:02    INFO  epoch 7 evaluating [time: 5.73s, valid_score: 0.027000]
26 Jan 03:02    INFO  valid result: 
recall@10 : 0.1003    mrr@10 : 0.027    ndcg@10 : 0.0437    hit@10 : 0.1003    precision@10 : 0.01    
26 Jan 03:03    INFO  epoch 8 training [time: 21.96s, train loss: 0.0003]
26 Jan 03:03    INFO  epoch 8 evaluating [time: 5.81s, valid_score: 0.027300]
26 Jan 03:03    INFO  valid result: 
recall@10 : 0.1015    mrr@10 : 0.0273    ndcg@10 : 0.0441    hit@10 : 0.1015    precision@10 : 0.0101    
26 Jan 03:03    INFO  epoch 9 training [time: 21.69s, train loss: 0.0002]
26 Jan 03:03    INFO  epoch 9 evaluating [time: 5.72s, valid_score: 0.027100]
26 Jan 03:03    INFO  valid result: 
recall@10 : 0.1017    mrr@10 : 0.0271    ndcg@10 : 0.044    hit@10 : 0.1017    precision@10 : 0.0102    
26 Jan 03:04    INFO  epoch 10 training [time: 21.87s, train loss: 0.0001]
26 Jan 03:04    INFO  epoch 10 evaluating [time: 5.84s, valid_score: 0.027700]
26 Jan 03:04    INFO  valid result: 
recall@10 : 0.0998    mrr@10 : 0.0277    ndcg@10 : 0.044    hit@10 : 0.0998    precision@10 : 0.01    
26 Jan 03:04    INFO  epoch 11 training [time: 21.68s, train loss: 0.0001]
26 Jan 03:04    INFO  epoch 11 evaluating [time: 5.69s, valid_score: 0.026700]
26 Jan 03:04    INFO  valid result: 
recall@10 : 0.098    mrr@10 : 0.0267    ndcg@10 : 0.0429    hit@10 : 0.098    precision@10 : 0.0098    
26 Jan 03:04    INFO  Finished training, best eval result in epoch 0
26 Jan 03:04    INFO  Loading model structure and parameters from saved/HGN-Jan-26-2022_02-59-10.pth
26 Jan 03:04    INFO  best valid : {'recall@10': 0.1005, 'mrr@10': 0.0304, 'ndcg@10': 0.0464, 'hit@10': 0.1005, 'precision@10': 0.01}
26 Jan 03:04    INFO  test result: {'recall@10': 0.1007, 'mrr@10': 0.0314, 'ndcg@10': 0.0473, 'hit@10': 0.1007, 'precision@10': 0.0101}
`

**Comment by chenyuwuxin** [comment]
@Li-Muyang Hello,thanks for your attention to RecBole!
I'm not sure what you mean by abnormal? Can you give a clearer explanation?
